File: easy/BackspaceCompare.cs
namespace Easy;

/// <summary>
///
/// </summary>
public class BackspaceCompare
{
    /// <summary>
    ///
    /// </summary>
    public static bool Solve(string s, string t)
    {
        return Process(s) == Process(t);
    }

    /// <summary>
    ///
    /// </summary>
    /// <param name="text"></param>
    /// <returns></returns>
    public static string Process(string text)
    {
        List<char> result = [];
        foreach (char c in text)
        {
            if (c == '#')
            {
                if (result.Count != 0)
                {
                    result.RemoveAt(result.Count - 1);
                }
            }
            else
            {
                result.Add(c);
            }
        }
        return new string([.. result]);
    }
}

File: easy/backspace_compare.go
package easy

func process(text string) string {
	var result []byte
	for i := 0; i < len(text); i++ {
		if text[i] == '#' {
			if len(result) > 0 {
				result = result[:len(result)-1]
			}
		} else {
			result = append(result, text[i])
		}
	}
	return string(result)
}
func BackspaceCompare(s, t string) bool {
	return process(s) == process(t)
}
